Unify LDA hyperparameter documentation

jonaschn opened the pull request:
Make documentation for alpha & eta consistent for AuthorTopicModel, LdaModel and LdaMulticore.

piskvorky commented on gensim/models/wrappers/ldamallet.py:
```diff
@@ -118,7 +126,16 @@ def __init__(self, mallet_path, corpus=None, num_topics=100, alpha=50, id2word=N
+            # Mallet's alpha parameter is the SumAlpha parameter
+            # (sum over topics of smoothing over doc-topic distributions. alpha_k = [SumAlpha] / [num topics])
+            # To keep the same interface with gensim, calculate the SumAlpha:
+            self.alpha = alpha * num_topics
```
I'd prefer to do it the other way around: keep `alpha` gensim-style internally, and convert to SumAlpha only at the point of Mallet API = when calling Mallet.

I think it's safer wrt expectations of gensim users. Is there a good reason to do it the other way around?

jonaschn replied:
No, you are totally right. 
It is indeed safer for gensim users.
I addressed your feedback and substantially improved the documentation.

piskvorky commented on gensim/models/wrappers/ldamallet.py:
```diff
@@ -87,8 +88,15 @@ def __init__(self, mallet_path, corpus=None, num_topics=100, alpha=50, id2word=N
+            (This may reduce the number of very small, poorly estimated topics,
+            but may disperse common words over several topics.)
+        eta : float, optional
+            Eta parameter of LDA (scalar for a symmetric prior over topic-word probability).
```
Where does the `0.01` default come from?

jonaschn replied:
It is the Mallet default behaviour (see [here](https://github.com/mimno/Mallet/blob/024a64ff9caf3f54586c5f70399da001f208305b/src/cc/mallet/topics/ParallelTopicModel.java#L93)):
`public static final double DEFAULT_BETA = 0.01;`

piskvorky replied:
I see. And this one is not a "sum", unlike alpha?

jonaschn replied:
You are right, as mentioned here #2903
Beta parameter: smoothing parameter for each topic-word. beta_w = [this value]

piskvorky replied:
Alright. So please change `alpha` to be stored internally as non-sum too (convert to sum only when needed) and we can merge. Thanks.

jonaschn commented:
@piskvorky I encorporated your feedback.

Unfortunately, the documentation build is failing. I am not familiar with sphinx and don't know how to fix this warning.

```
Warning, treated as error:
/home/circleci/gensim/gensim/models/atmodel.py:docstring of gensim.models.atmodel.AuthorTopicModel:29:Block quote ends without a blank line; unexpected unindent.
make: *** [Makefile:37: html] Error 2
make: Leaving directory '/home/circleci/gensim/docs/src'
ERROR: InvocationError for command /usr/bin/make -C docs/src clean html (exited with code 2)
```

jonaschn commented:
As gensim dropped the support for the Mallet wrapper, I removed this part of the PR.
You can find the improved and fixed (!) version of the Mallet wrapper for LDA and Labeled LDA here:
https://github.com/jonaschn/gensim/tree/labeled-lda
